meteo/_hycom.py

```diff
@@ -0,0 +1,130 @@
+import logging
+
+import numpy as np
+import xarray as xr
+import seawater as sw
+from time import time
+
+import pandas as pd
+
+from ._constants import HYCOM_MAX_DEPTH_INDEX
+from ._constants import POTENTIAL_TEMP_CORRECTION
+from ._constants import Bbox
+
+logger = logging.getLogger(__name__)
+
+def ConvertTemp(salt, temp, dep):
+    # nz = temp.shape[0]
+    # ny = temp.shape[1]
+    # nx = temp.shape[2]
+    pr = np.ones(temp.shape)
+    pre = pr*dep[:,None, None]
+    Pr = np.zeros(temp.shape)
+    ptemp = sw.ptmp(salt, temp, pre, Pr)*POTENTIAL_TEMP_CORRECTION
+    return ptemp
+
+
+def get_database(date, Bbox=None):
+    if date >= pd.Timestamp(2018, 12, 4):
+        database = 'GLBy0.08/expt_93.0'
+    elif date >= pd.Timestamp(2018, 1, 1) and date < pd.Timestamp(2018, 12, 4):
+        database = 'GLBv0.08/expt_93.0'
+    elif date >= pd.Timestamp(2017, 10, 1) and date < pd.Timestamp(2018, 1, 1):
+        database = 'GLBv0.08/expt_92.9'
+    elif date >= pd.Timestamp(2017, 6, 1) and date < pd.Timestamp(2017, 10, 1):
+        database = 'GLBv0.08/expt_57.7'
+    elif date >= pd.Timestamp(2017, 2, 1) and date < pd.Timestamp(2017, 6, 1):
+        database = 'GLBv0.08/expt_92.8'
+    elif date >= pd.Timestamp(2016, 5, 1) and date < pd.Timestamp(2017, 2, 1):
+        database = 'GLBv0.08/expt_57.2'
+    elif date >= pd.Timestamp(2016, 1, 1) and date < pd.Timestamp(2016, 5, 1):
+        database = 'GLBv0.08/expt_56.3'
+    elif date >= pd.Timestamp(1994, 1, 1) and date < pd.Timestamp(2016, 1, 1):
+        database = f'GLBv0.08/expt_53.X/data/{date.year}'
+    else:
+        raise ValueError(f'No data fro {date}!')
+    return database
+
+
+def get_idxs(date, database, bbox):
+    if date >= pd.Timestamp.now():
+        raise Exception("select time before today")
+    else:
+        baseurl=f'https://tds.hycom.org/thredds/dodsC/{database}?lat[0:1:-1],lon[0:1:-1],time[0:1:-1],depth[0:1:-1]'
+
+    ds = xr.open_dataset(baseurl)
+    lon = ds.lon.data
+    lat = ds.lat.data
+    times = ds.time.data
+    ds.close()
+
+    time_idx = np.where(date == times)[0]
+    if len(time_idx) == 0:
+        raise Exception(f"No data for this date: {date}")
+    
+    lat_idxs=np.where((lat>=bbox.ymin-0.5)&(lat<=bbox.ymax+0.5))[0]
+    lon_idxs=np.where((lon>=bbox.xmin-0.5) & (lon<=bbox.xmax+0.5))[0]
+
+    return time_idx[0], lon_idxs[0], lon_idxs[-1], lat_idxs[0], lat_idxs[-1]
+
+
+def download_hycom(year, month, day, rnday=1, bnd=True, nudge=False, sub_sample=1, output_path=None):
```

Comment:
Lots of arguments here, put one argument per line and add type hints since this function is going to be part of the public API.